### ISSUE TITLE
`CREATE DATA ref LIKE LINE OF data` on a generic `TYPE data` parameter is not reported

### ISSUE THREAD

**Issue by oisee** [abaplint/abaplint#4332]
Hi Lars,

```abap
CLASS zcl_x DEFINITION PUBLIC.
  PUBLIC SECTION.
    CLASS-METHODS m CHANGING data TYPE data.
ENDCLASS.
CLASS zcl_x IMPLEMENTATION.
  METHOD m.
    DATA ref TYPE REF TO data.
    CREATE DATA ref LIKE LINE OF data.
  ENDMETHOD.
ENDCLASS.
```

@abaplint/core 2.120.59 reports no syntax error. On a 7.5x system the class does not activate:

> "DATA" is not an internal table.

The form the system accepts goes through a field symbol typed as a table:

```abap
FIELD-SYMBOLS <at> TYPE ANY TABLE.
ASSIGN data TO <at>.
CREATE DATA ref LIKE LINE OF <at>.
```

Found in open-abap-core's `/UI2/CL_JSON=>_DESERIALIZE`, which has this line; a PR for that side is separate.
